pandas/core/series.py

```diff
@@ -1131,6 +1131,239 @@ def _set_value(self, label, value, takeable: bool = False):
+        dtype: int64
+        """
+        inplace = validate_bool_kwarg(inplace, "inplace")
+        if not isinstance(expr, str):
+            msg = f"expr must be a string to be evaluated, {type(expr)} given"
+            raise ValueError(msg)
+        kwargs["level"] = kwargs.pop("level", 0) + 1
+        kwargs["target"] = None
+        res = self.eval(expr, **kwargs)
+
+        try:
+            new_data = self.loc[res]
+        except ValueError:
+            # when res is multi-dimensional loc raises, but this is sometimes a
+            # valid query
+            new_data = self[res]
+
+        if inplace:
+            self._update_inplace(new_data)
+        else:
+            return new_data
+
+    def eval(self, expr, inplace=False, **kwargs):
+        """
+        Evaluate a string describing operations on Series columns.
+
+        Operates on columns only, not specific rows or elements.  This allows
+        `eval` to run arbitrary code, which can make you vulnerable to code
+        injection if you pass user input to this function.
+
+        Parameters
+        ----------
+        expr : str
+            The expression string to evaluate.
+        inplace : bool, default False
+            If the expression contains an assignment, whether to perform the
+            operation inplace and mutate the existing Series. Otherwise,
+            a new Series is returned.
+        **kwargs
+            See the documentation for :func:`eval` for complete details
+            on the keyword arguments accepted by
+            :meth:`~pandas.Series.query`.
+
+        Returns
+        -------
+        ndarray, scalar, or pandas object
+            The result of the evaluation.
+
+        See Also
+        --------
+        Series.query : Evaluates a boolean expression to query the columns
+            of a frame.
+        eval : Evaluate a Python expression as a string using various
+            backends.
+
+        Notes
+        -----
+        For more details see the API documentation for :func:`~eval`.
+        For detailed examples see :ref:`enhancing performance with eval
+        <enhancingperf.eval>`.
+
+        Examples
+        --------
+        >>> series = pd.Series(1, index=range(5))
+        >>> series
+        0    1
+        1    1
+        2    1
+        3    1
+        4    1
+        dtype: int64
+
+        Assignment is allowed though by default the original DataFrame is not
+        modified.
+
+        >>> op = f"1.2 {'+'} index"
+        >>> series.eval(op)
+        0    1.2
+        1    2.2
+        2    3.2
+        3    4.2
+        4    5.2
+        dtype: float64
+
+        Use ``inplace=True`` to modify the original DataFrame.
+
+        >>> dict1 = {"a": 1}
+        >>> dict2 = {"b": 2}
+        >>> series.eval("c = a + b", inplace=True, resolvers=[dict1, dict2])
+        >>> series
+        0    1
+        1    1
+        2    1
+        3    1
+        4    1
+        c    3
+        dtype: int64
```

Comment:
Maybe instead of this example we can do something simpler like `index > 5`?

Reply:
I see what you mean, originally i wanted to inlclude it so that we can show how "inplace" works, to be consistent with what was under eval() doc under frame.py

Reply:
Oh, I see. That's cool then, we can stick with this example